src/Lint.jl

```diff
@@ -350,34 +356,116 @@ function lintdir{T<:AbstractString}(dir::T, ctx::LintContext=LintContext())
     ctx.messages
 end
 
-function readandwritethestream(conn)
-    # println("Connection accepted")
-    # Get file, code length and code
-    file = strip(readline(conn))
-    # println("file: ", file)
-    code_len = parse(Int, strip(readline(conn)))
-    # println("Code bytes: ", code_len)
-    code = Compat.UTF8String(read(conn, code_len))
-    # println("Code received")
-    # Do the linting
-    msgs = lintfile(file, code)
-    # Write response to socket
-    for i in msgs
-        write(conn, string(i))
+function convertmsgtojson(msgs, style, dict_data)
+    if style == "lint-message"
+        return JSON.json(msgs)
+    end
+    output = []
+    for msg in msgs
+        evar = msg.variable
+        txt = msg.message
+        file = msg.file
+        linenumber = msg.line
+        # Atom index starts from zero thus minus one
+        errorrange = Array[[linenumber-1, 0], [linenumber-1, 80]]
+        code = string(msg.code)
+        if code[1] == 'I'
+            etype = "info"
+            etypenumber = 3
+        elseif code[1] == 'W'
+            etype = "warning"
+            etypenumber = 2
+        else
+            etype = "error"
+            etypenumber = 1
+        end
+
+        if style == "standard-linter-v1"
+            if haskey(dict_data,"show_code")
+                if !dict_data["show_code"]
+                    msgtext = "$evar: $txt"
+                end
+            else
+                msgtext = "$code $evar: $txt"
+            end
+            push!(output, Dict("type" => etype,
+                               "text" => msgtext,
+                               "range" => errorrange,
+                               "filePath" => file))
+        elseif style == "vscode"
+            push!(output, Dict("severity" => etypenumber,
+                               "message" => "$evar: $txt",
+                               "range" => errorrange,
+                               "filePath" => file,
+                               "code" => code,
+                               "source" => "Lint.jl"))
+        elseif style == "standard-linter-v2"
+            push!(output, Dict("severity" => etype,
+                               "location" => Dict("file" => file,
+                                                   "position" => errorrange),
+                               "excerpt" => code,
+                               "description" => "$evar: $txt"))
+
+        end
+    end
+    return JSON.json(output)
```

Comment:
Actually I think it might be better to just return the `output` here, and handle the vector of dicts in the other function. Then we can avoid allocating an intermediate string and print directly to the stream, which might be faster. It also becomes easier to test this function if we want to add unit tests for more styles in the future.